Nodejs/Product/Nodejs/SharedProject/CommonProjectNode.cs

```diff
@@ -801,20 +747,17 @@ protected override ReferenceContainerNode CreateReferenceContainerNode()
             return new CommonReferenceContainerNode(this);
         }
 
-        private void FileNameChanged(object sender, RenamedEventArgs e)
-        {
-            // we just generate a delete and creation here - we're just updating the hierarchy
-            // either changing icons or updating the non-project elements, so we don't need to
-            // generate rename events or anything like that.  This saves us from having to 
-            // handle updating the hierarchy in a special way for renames.
-            this.QueueFileSystemChanges(
-                new FileSystemChange(this, WatcherChangeTypes.Deleted, e.OldFullPath, isRename: true),
-                new FileSystemChange(this, WatcherChangeTypes.Created, e.FullPath, isRename: true));
-        }
-
-        private void FileExistanceChanged(object sender, FileSystemEventArgs e)
+        private void FolderChangedOnDisk(object sender, FolderChangedEventArgs e)
         {
-            this.QueueFileSystemChanges(new FileSystemChange(this, e.ChangeType, e.FullPath));
+            var file = e.FileName;
+            if (File.Exists(file) || Directory.Exists(file))
```

Comment:
>if (File.Exists(file) || Directory.Exists(file)) [](start = 12, length = 48)

Why do we have to hit the disk here?  Is this the event you mentioned was inadequate?

Reply:
Yep, they currently don't report what changed to the file/folder. Hopefully in the v2 of the API. Note this is all on BG thread, so it's not as bad as it seems.